app/models/shipit/stack.rb

```diff
@@ -179,19 +181,23 @@ def head
     end
 
     def merge_status
-      if locked?
-        'locked'
+      return 'locked' if locked?
+      return 'failure' if branch_status == 'failure'
+      if undeployed_commits_count > (maximum_commits_per_deploy || DEFAULT_MAX_COMMITS_PER_DEPLOY) * 1.5
```

Comment:
Do you think it would be worthwhile to override `maximum_commits_per_deploy` to have the default there?